src/createPackage/models/createPackageManager.ts

```diff
@@ -15,26 +17,91 @@ export class CreatePackageManager {
     @inject(RasterCatalogManagerClient) private readonly rasterCatalogManager: RasterCatalogManagerClient
   ) {}
 
-  public async createPackage(userInput: ICreatePackage): Promise<IJobCreationResponse> {
+  public async createPackage(userInput: ICreatePackage): Promise<ICreateJobResponse | ICallbackResponse> {
     const layer = await this.rasterCatalogManager.findLayer(userInput.dbId);
-
     const layerMetadata = layer.metadata;
 
-    const workerInput: IWorkerInput = {
-      footprint: layerMetadata.footprint as Polygon | MultiPolygon,
-      bbox: userInput.bbox,
-      version: layerMetadata.productVersion as string,
-      cswProductId: layerMetadata.productId as string,
-      targetResolution: userInput.targetResolution,
-      dbId: userInput.dbId,
-      packageName: userInput.packageName,
-      callbackURL: userInput.callbackURL,
-      tilesPath: (layerMetadata.productId as string) + sep + (layerMetadata.productVersion as string),
-      priority: userInput.priority ?? DEFAULT_PRIORITY,
-      crs: userInput.crs ?? DEFAULT_CRS,
+    const { productId: resourceId, productVersion: version, footprint, productType } = layerMetadata;
+    const { bbox, dbId, targetResolution, crs, priority, callbackURL } = userInput;
+
+    const dupParams: JobDuplicationParams = {
+      resourceId: resourceId as string,
+      version: version as string,
+      dbId,
+      targetResolution,
```

Comment:
i think we should determine the zoom level and compare it to prevent similar resolutions with the same "zoom level" result